Split off `KeyboardHandle::filter` callback to separate method

ids1024 opened the pull request:
Requires https://github.com/Smithay/smithay/pull/1533. If the `input_intercept` changes there are controversial, this could instead use a a few more arguments, or something involving `input_intercept`/`input_forward` instead of `input` (but that's probably not simpler).

https://github.com/pop-os/cosmic-comp/pull/465 will require some additional logic here to filter out the key grabs. I don't really want this to get *more* complicated so a wanted to try splitting things up first. Just moving the callback to a method is a start, though more could be improved.

Drakulix reviewed (commented):
Seems sensible to me

ids1024 commented:
Updated now that Smithay changes are merged, and added a commit removing the `needs_key_repetition` argument which should be unneeded with https://github.com/Smithay/smithay/pull/1535.
